src/Components/WorkloadRules/WorkloadRules.js

```diff
@@ -104,48 +134,158 @@ const WorkloadRules = ({ workload }) => {
-                  tooltipProps={{ position: TooltipPosition.bottom }}
-                /> */}
-              </div>
-            ),
-          },
-        ],
-      },
-      {
-        cells: [
-          {
-            title: <ExpandedRulesDetails recommendations={recommendations} />,
-          },
-        ],
-      },
-    ]);
+  const pruneFilters = (localFilters, filterCategories) => {
+    const prunedFilters = Object.entries(localFilters);
+    return prunedFilters.length > 0
+      ? prunedFilters.reduce((arr, item) => {
+          if (filterCategories[item[0]]) {
+            const category = filterCategories[item[0]];
+            const chips = Array.isArray(item[1])
+              ? item[1].map((value) => {
+                  const selectedCategoryValue = category.values.find(
+                    (values) => values.value === String(value)
+                  );
+                  return selectedCategoryValue
+                    ? {
+                        name:
+                          selectedCategoryValue.text ||
+                          selectedCategoryValue.label,
+                        value,
+                      }
+                    : { name: value, value };
+                })
+              : [
+                  {
+                    name: category.values.find(
+                      (values) => values.value === String(item[1])
+                    ).label,
+                    value: item[1],
+                  },
+                ];
+            return [
+              ...arr,
+              {
+                category: capitalize(category.title),
+                chips,
+                urlParam: category.urlParam,
+              },
+            ];
+          } else if (item[0] === 'description') {
+            return [
+              ...arr,
+              ...(item[1].length > 0
+                ? [
+                    {
+                      category: 'Description',
+                      chips: [{ name: item[1], value: item[1] }],
+                      urlParam: item[0],
+                    },
+                  ]
+                : []),
+            ];
+          } else if (item[0] === 'object_id') {
+            return [
+              ...arr,
+              ...(item[1].length > 0
+                ? [
+                    {
+                      category: 'Object ID',
+                      chips: [{ name: item[1], value: item[1] }],
+                      urlParam: item[0],
+                    },
+                  ]
+                : []),
+            ];
+          } else {
+            return arr;
+          }
+        }, [])
+      : [];
+  };
+
+  const buildFilterChips = () => {
+    const localFilters = { ...filters };
+    delete localFilters.sortIndex;
+    delete localFilters.sortDirection;
+    return pruneFilters(localFilters, WORKLOAD_RULES_FILTER_CATEGORIES);
+  };
+
+  const activeFiltersConfig = {
+    showDeleteButton: filtersApplied ? true : false,
```

Comment:
Can we get rid of that? The toolbar should be able to decide whether to render the reset filters button or not internally. If the toolbar shows the reset button, then we most probably pass non-empty array to `filters`. If there are no filters applied, then `buildFiltersChips` should return an empty array so that the function is semantic.

Reply:
No we can't, we tried to make it work together with Sebastian and this is the solution we came up with